zerver/lib/actions.py

```diff
@@ -2426,6 +2426,10 @@ def do_update_message(user_profile, message_id, subject, propagate_mode, content
         event["content"] = content
         event["rendered_content"] = rendered_content
 
+        prev_content = edit_history_event['prev_content']
+        if Message.content_has_attachment(prev_content) or Message.content_has_attachment(message.content):
+            check_attachment_reference_change(prev_content, message)
+
```

Comment:
This isn't a great place to put this code, because there will be update consistency issues of the user submitted their change with an empty topic.  and thus the JsonableError about 10 lines down is thrown.  Probably the right fix is to just move that piece of validation, though; I'll just do that.